### ISSUE TITLE
BotDispatch refresh tries to use version "Dispatch" instead of version specified in CLA using -v

### ISSUE THREAD

**Issue by Wilburre** [Microsoft/botbuilder-tools#918]
## Tool
Name: Dispatch (refresh command)
Version: 1.2.2 (node version 10.9.0)
OS: Windows 10 Enterprise Ver 1803 Build 17134.523

## Describe the bug
Refreshing a LUIS dispatch model/app using dispatch refresh -v <myver i.e. 1.7> using the only .dispatch file in the working directory. Instead of using the version number passed in it uses the version "Dispatch" which allows the model to be built and imported, but not activated, trained or published unless done manually via luis portal.

## To Reproduce
1. Have a dispatch file for an existing dispatch model
2. Set the versions for the services in the file
3. Run dispatch refresh -v 1.7 from bash in the directory containing the .dispatch
4. Let it error out

## Expected behavior
Expect dispatch to import, activate, train and publish the version of the dispatch app passed in via cli i.e. dispatch refresh -v '1.7' - should use 1.7 not "Dispatch" which doesn't exist.

[bug]
![dispatch file](https://user-images.githubusercontent.com/21064706/51718398-2c7d3900-2099-11e9-9f74-93c84889fd97.PNG)
![dispatch output](https://user-images.githubusercontent.com/21064706/51718590-0a37eb00-209a-11e9-9715-34fe42e03a94.png)

**Comment by sgellock** [comment]
@cahann you should make the call whether this is a 4.3 or a 4.4 issue for you

**Comment by cahann** [comment]
will target 4.3

**Comment by tsuwandy** [comment]
@Wilburre , did you manually update the Dispatch app version in .dispatch file?  Then also pass in the version parameter for fresh?  "dispatch refresh -v {new version}" works fine for me.  I didn't do any update to the .dispatch file.

**Comment by Wilburre** [comment]
@tsuwandy No, I only updated the version number for my 2 services and left the dispatch ver number alone as the dispatch refresh operation when done successfully seems to mutate the file to the version number passed in with -v.

TLDR - passed in desired new dispatch version number with -v and adjusted the service version numbers manually in the .dispatch.

I've looked into this further from my side and if a version of the dispatch exists in LUIS that is called "Dispatch" the refresh is successful - However it ends up importing the version created and specified with -v, but still tries to train and publish the version "Dispatch" - so I can confirm in my situation when training and publishing dispatch refresh seems to be ignoring -v.

If it provides more information the version numbers I am using are of the form "1.x" i.e. 1.6, 1.7 etc.

**Comment by tsuwandy** [comment]
oh, got it.  what version of dispatch do you have? "dispatch -v"

I was able to pass in whatever version to "dispatch refresh -v {new version}" but the new version is not published.  This is the bug, correct? Thanks!

**Comment by Wilburre** [comment]
@tsuwandy  my organisation maintains several dispatch apps in LUIS running different version numbers and have been seeing this issue for a couple of weeks. I get the issue on dispatch 1.2.2 and previously 1.2.1. The version numbers for my dispatch apps currently are 1.9 (the one we're discussing though all of them suffer this issue on refresh), 1.9, 0.2 and "Dispatch" - the one called dispatch works flawlessly but it was our first foray into dispatch apps and isn't versioned - just overwritten each time we change it.

The bug I seem to get is the new version gets imported, but not activated, trained or published. If you take a look in the screenshot of the CLI error the dispatch refresh is trying to access the training url for the version "Dispatch" of the dispatch app in question - i.e. it is failing at activation/training and publishing.

**Comment by tsuwandy** [comment]
Yes, I think I get the bug, the new version is created, but not trained and published.  I'm working on the fix at the moment.  Thanks!

**Comment by Wilburre** [comment]
@tsuwandy thanks mate

**Comment by tsuwandy** [comment]
Fix is ready, will publish sometime tomorrow.  There is an optional to publish to staging slot too.  “-publishToStaging true”.  Thanks for reporting this!

**Comment by tsuwandy** [comment]
update published to npmjs

**Comment by Wilburre** [comment]
@tsuwandy thanks for quick fix - tested it this morning and behaves as expected on 1.2.3.
